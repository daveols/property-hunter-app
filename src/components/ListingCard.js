import React from 'react';
import {Platform, StyleSheet} from 'react-native';
import {Card, Caption, Colors} from 'react-native-paper';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';

const ListingCard = ({
  propertyDetails: {streetNumber, street, suburb, bedrooms, bathrooms},
  media,
}) => {
  const imageUri =
    media[0] && media[0].category === 'Image' ? media[0].url : undefined;
  return (
    <Card style={styles.card}>
      <Card.Cover source={{uri: imageUri}} />
      <Card.Title title={`${streetNumber} ${street}`} subtitle={suburb} />
      {Platform.OS === 'ios' && (
        <Card.Content style={styles.content}>
          <MaterialCommunityIcon
            name="bed-empty"
            size={16}
            color={Colors.grey500}
            style={styles.icon}
          />
          <Caption style={styles.caption}>{bedrooms}</Caption>
          <MaterialCommunityIcon
            name="water-pump"
            size={16}
            color={Colors.grey500}
            style={styles.icon}
          />
          <Caption style={styles.caption}>{bathrooms}</Caption>
        </Card.Content>
      )}
    </Card>
  );
};

const styles = StyleSheet.create({
  card: {
    margin: 16,
    marginVertical: 8,
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    marginRight: 4,
  },
  caption: {
    marginRight: 16,
  },
});

export default ListingCard;
