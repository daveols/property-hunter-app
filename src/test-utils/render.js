import React from 'react';
import {render} from '@testing-library/react-native';

import {DomainAuthProvider} from 'contexts/DomainAuthContext';

export default (ui, {...options} = {}) => {
  const Providers = ({children}) => (
    <DomainAuthProvider>{children}</DomainAuthProvider>
  );

  return {...render(ui, {wrapper: Providers, ...options})};
};
