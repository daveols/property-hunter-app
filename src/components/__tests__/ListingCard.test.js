import React from 'react';
import {Platform} from 'react-native';

import render from 'test-utils/render';

import ListingCard from 'components/ListingCard';

const mockListing = {
  propertyDetails: {
    streetNumber: '69',
    street: 'Hosier Lane',
    suburb: 'Melbourne',
    bedrooms: 6,
    bathrooms: 9,
  },
  media: [
    {
      category: 'Image',
      url: 'https://sweetmelbournepics.com.au/hosier-lane.png',
    },
  ],
};

describe('ListingCard', () => {
  test('renders listing correctly - ios', () => {
    const {container, queryByText} = render(<ListingCard {...mockListing} />);

    expect(queryByText('69 Hosier Lane')).toBeTruthy();
    expect(queryByText('Melbourne')).toBeTruthy();
    expect(queryByText('6')).toBeTruthy();
    expect(queryByText('9')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });

  test('renders listing correctly - android', () => {
    Platform.OS = 'android';
    const {container, queryByText} = render(<ListingCard {...mockListing} />);

    expect(queryByText('69 Hosier Lane')).toBeTruthy();
    expect(queryByText('Melbourne')).toBeTruthy();
    expect(queryByText('6')).toBeNull();
    expect(queryByText('9')).toBeNull();

    expect(container).toMatchSnapshot();
  });
});
