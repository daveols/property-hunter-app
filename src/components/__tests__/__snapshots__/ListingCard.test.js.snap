// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCard renders listing correctly - android 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "elevation": 1,
        "margin": 16,
        "marginVertical": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <TouchableWithoutFeedback
      delayPressIn={0}
      disabled={true}
    >
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              Object {
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              undefined,
            ]
          }
        >
          <Image
            siblings={
              Array [
                "withTheme(Card.Cover)",
                "withTheme(Card.Title)",
              ]
            }
            source={
              Object {
                "uri": "https://sweetmelbournepics.com.au/hosier-lane.png",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "resizeMode": "cover",
                  "width": undefined,
                },
                Object {
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "height": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 50,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              69 Hosier Lane
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              Melbourne
            </Text>
          </View>
          <View />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;

exports[`ListingCard renders listing correctly - ios 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "elevation": 1,
        "margin": 16,
        "marginVertical": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <TouchableWithoutFeedback
      delayPressIn={0}
      disabled={true}
    >
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              Object {
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              undefined,
            ]
          }
        >
          <Image
            siblings={
              Array [
                "withTheme(Card.Cover)",
                "withTheme(Card.Title)",
                "Card.Content",
              ]
            }
            source={
              Object {
                "uri": "https://sweetmelbournepics.com.au/hosier-lane.png",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "resizeMode": "cover",
                  "width": undefined,
                },
                Object {
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              Object {
                "height": 72,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "height": 50,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                    Array [
                      Object {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "minHeight": 30,
                        },
                        Object {
                          "marginBottom": 0,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              69 Hosier Lane
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                    Array [
                      Object {
                        "fontSize": 12,
                        "letterSpacing": 0.4,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "marginVertical": 0,
                          "minHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ],
                ]
              }
            >
              Melbourne
            </Text>
          </View>
          <View />
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingBottom": 16,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#9e9e9e",
                  "fontSize": 16,
                },
                Object {
                  "marginRight": 4,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginRight": 16,
                    },
                  ],
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginRight": 16,
                    },
                  ],
                ],
              ]
            }
          >
            6
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#9e9e9e",
                  "fontSize": 16,
                },
                Object {
                  "marginRight": 4,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginRight": 16,
                    },
                  ],
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginRight": 16,
                    },
                  ],
                ],
              ]
            }
          >
            9
          </Text>
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;
